src/TorSharp/Tools/DirectoryUtility.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.IO;
+
+namespace Knapcode.TorSharp.Tools
+{
+    internal static class DirectoryUtility
+    {
+        public static void CreateDirectoryIfNotExists(params string[] path)
+        {
+            foreach (var p in path)
+                CreateDirectoryIfNotExists(p);
+        }
+
+        public static void CreateDirectoryIfNotExists(string path)
+        {
+            if (!Directory.Exists(path))
+                Directory.CreateDirectory(path);
```

Comment:
`Directory.CreateDirectory` already gracefully handles the case when the directory already exists. I think this `CreateDirectoryIfNotExists` method is not necessary.